gtsfm/frontend/verifier/degensac.py

```diff
@@ -18,87 +18,63 @@
             camera_intrinsics_i1: intrinsics for image #i1.
             camera_intrinsics_i2: intrinsics for image #i2.
+            use_intrinsics_in_verification (optional): Flag to perform keypoint normalization and compute the essential
+                                                       matrix instead of fundamental matrix. This should be preferred
+                                                       when the exact intrinsics are known as opposed to approximating
+                                                       them from exif data. Defaults to False.
 
         Returns:
             Estimated rotation i2Ri1, or None if it cannot be estimated.
             Estimated unit translation i2Ui1, or None if it cannot be estimated.
             Indices of verified correspondences, of shape (N, 2) with N <= N3. These are subset of match_indices.
         """
-        print("WARNING: Degensac verifier cannot compute essential matrix directly using exact intrinsics")
-
-        return self.verify_with_approximate_intrinsics(
-            keypoints_i1,
-            keypoints_i2,
-            match_indices,
-            camera_intrinsics_i1,
-            camera_intrinsics_i2,
-        )
-
-    def verify_with_approximate_intrinsics(
-        self,
-        keypoints_i1: Keypoints,
-        keypoints_i2: Keypoints,
-        match_indices: np.ndarray,
-        camera_intrinsics_i1: Cal3Bundler,
-        camera_intrinsics_i2: Cal3Bundler,
-    ) -> Tuple[Optional[Rot3], Optional[Unit3], np.ndarray]:
-        """Estimates the essential matrix and verifies the feature matches.
+        if use_intrinsics_in_verification:
+            logger.warning("Essential Matrix estimation not supported by DEGENSAC verifier")
 
-        Note: this function is preferred when camera intrinsics are approximate (i.e from image size/exif). The feature
-        coordinates are used to compute the fundamental matrix, which is then converted to the essential matrix.
-
-        Args:
-            keypoints_i1: detected features in image #i1.
-            keypoints_i2: detected features in image #i2.
-            match_indices: matches as indices of features from both images, of shape (N3, 2), where N3 <= min(N1, N2).
-            camera_intrinsics_i1: intrinsics for image #i1.
-            camera_intrinsics_i2: intrinsics for image #i2.
-
-        Returns:
-            Estimated rotation i2Ri1, or None if it cannot be estimated.
-            Estimated unit translation i2Ui1, or None if it cannot be estimated.
-            Indices of verified correspondences, of shape (N, 2) with N <= N3. These are subset of match_indices.
-        """
         verified_indices = np.array([], dtype=np.uint32)
 
         # check if we don't have the minimum number of points
-        if match_indices.shape[0] < self.min_pts:
+        if match_indices.shape[0] < self.min_pts_f_matrix:
```

Comment:
why is this an argument, rather than a global var?

Reply:
each function might have a different number of minimum points requirement?

Reply:
stick with 5 and 8 here.

Reply:
done